pkg/ctl/install/fluxinstaller.go

```diff
@@ -0,0 +1,558 @@
+		if cleanCloneDir {
+			_ = fi.gitClient.DeleteLocalRepo()
+		} else {
+			logger.Critical("You may find the local clone of %s used by eksctl at %s",
+				fi.opts.gitURL,
+				cloneDir)
+		}
+	}()
+
+	logger.Info("Writing Flux manifests")
+	fluxManifestDir := filepath.Join(cloneDir, fi.opts.gitFluxPath)
+	if err := writeFluxManifests(fluxManifestDir, manifests); err != nil {
+		return err
+	}
+
+	if fi.opts.amend {
+		logger.Info("Stopping to amend the the Flux manifests, please exit the shell when done.")
+		if err := runShell(fluxManifestDir); err != nil {
+			return err
+		}
+		// Re-read the manifests, as they may have changed:
+		manifests, err = readFluxManifests(fluxManifestDir)
+		if err != nil {
+			return err
+		}
+	}
+
+	logger.Info("Applying manifests")
+	if err := fi.applyManifests(manifests); err != nil {
+		return err
+	}
+
+	if len(secrets) > 0 {
+		logger.Info("Applying Helm TLS Secret(s)")
+		if err := fi.applySecrets(secrets); err != nil {
+			return err
+		}
+		logger.Warning("Note: certificate secrets aren't added to the Git repository for security reasons")
+	}
+
+	if !fi.opts.noHelmOp {
+		logger.Info("Waiting for Helm Operator to start")
+		if err := waitForHelmOpToStart(ctx, fi.opts.namespace, fi.opts.timeout, fi.k8sRestConfig, fi.k8sClientSet); err != nil {
+			return err
+		}
+		logger.Info("Helm Operator started successfully")
+	}
+
+	logger.Info("Waiting for Flux to start")
+	fluxSSHKey, err := waitForFluxToStart(ctx, fi.opts.namespace, fi.opts.timeout, fi.k8sRestConfig, fi.k8sClientSet)
+	if err != nil {
+		return err
+	}
+	logger.Info("Flux started successfully")
+
+	logger.Info("Committing and pushing manifests to %s", fi.opts.gitURL)
+	if err := fi.addFilesToRepo(ctx, cloneDir); err != nil {
+		return err
+	}
+	cleanCloneDir = true
+
+	logger.Info("Flux will only operate properly once it has write-access to the Git repository")
+	logger.Info("please configure %s so that the following Flux SSH public key has write access to it\n%s",
+		fi.opts.gitURL, fluxSSHKey.Key)
+	return nil
+}
+
+func (fi *fluxInstaller) setupPKI() (*publicKeyInfrastructure, *publicKeyInfrastructurePaths, error) {
+	if fi.opts.noHelmOp {
+		return nil, nil, nil
+	}
+
+	var err error
+	if fi.opts.noTiller {
+		logger.Info("Configuring TLS for Helm Operator using files from --helmop-tls-*-path flags")
+		pkiPaths := &publicKeyInfrastructurePaths{
+			caCertificate:     fi.opts.helmOpTLSCACertFile,
+			clientKey:         fi.opts.helmOpTLSKeyFile,
+			clientCertificate: fi.opts.helmOpTLSCertFile,
+		}
+		var pki = &publicKeyInfrastructure{}
+		if err = pki.loadFrom(pkiPaths); err != nil {
+			return nil, nil, err
+		}
+		return pki, pkiPaths, nil
+	}
+
+	logger.Info("Generating public key infrastructure for the Helm Operator and Tiller")
+	logger.Info("  this may take up to a minute, please be patient")
+	fi.opts.tillerHost = tillerServiceName + "." + fi.opts.tillerNamespace
+	pki, err := newPKI(fi.opts.tillerHost, 5*365*24*time.Hour, 4096)
```

Comment:
If we create this here instead of using dependency injection with an interface it will be hard to mock, or how will flux installer be tested?

Reply:
Uhm, do we need to mock it? That function doesn't really need any explicit I/O

Reply:
don't we call something about reading or writing the certs to the filesystem? if there is no IO in the PKI package then I don't see a strong need for it to be mockable

Reply:
Ah, that's done when `saveTo()`, yes. Uhm, let me think about it.

Reply:
I tried to use `afero` but  I couldn't, because `tillerinstall.Secret()` expects a real file.

Reply:
@2opremio perhaps we should have an interface for PKI that we can mock later then?